src/arrow1/reader_async.rs

```diff
@@ -1,15 +1,670 @@
+                let meta_range = (file_size - metadata_byte_length - 8)..file_size;
+
+                let meta_bytes = self.get_bytes(meta_range).await.unwrap();
+
+                decode_metadata(&meta_bytes[0..meta_bytes.len() - 8])?
+            } else {
+                let metadata_start = suffix_len - metadata_byte_length - 8;
+
+                let slice = &suffix[metadata_start..suffix_len - 8];
+                decode_metadata(slice)?
+            };
+            Ok(Arc::new(meta))
+        }
+        .boxed()
+    }
+}
+
+pub async fn make_range_request_with_client(
+    url: String,
+    client: Client,
+    range_str: String,
+) -> std::result::Result<Bytes, JsValue> {
+    let (sender, receiver) = oneshot::channel();
+    spawn_local(async move {
+        let resp = client
+            .get(url)
+            .header("Range", range_str)
+            .send()
+            .await
+            .unwrap()
+            .error_for_status()
+            .unwrap();
+        let bytes = resp.bytes().await.unwrap();
+        sender.send(bytes).unwrap();
+    });
+    let data = receiver.await.unwrap();
+    Ok(data)
+}
+
+// async fn make_request(
+//     url: &str,
+//     client: &Client,
+//     range: Range<usize>,
+// ) -> parquet::errors::Result<Bytes> {
+//     todo!()
+// }
+
+// async fn get_bytes<'a>(
+//     url: &'a str,
+//     client: &'a Client,
+//     range: Range<usize>,
+// ) -> BoxFuture<'a, parquet::errors::Result<Bytes>> {
+//     async move {
+//         let range_str =
+//             range_from_start_and_length(range.start as u64, (range.end - range.start) as u64);
+
+//         // Map reqwest error to parquet error
+//         let map_err = |err| parquet::errors::ParquetError::External(Box::new(err));
+
+//         let resp = client
+//             .get(url)
+//             .header("Range", range_str)
+//             .send()
+//             .await
+//             .map_err(map_err)?
+//             .error_for_status()
+//             .map_err(map_err)?;
+//         let bytes = resp.bytes().await.map_err(map_err)?;
+//         Ok(bytes)
+//     }
+//     .boxed()
+// }
+
+// async fn get_byte_ranges<'a>(
+//     url: &'a str,
+//     client: &'a Client,
+//     ranges: Vec<Range<usize>>,
+//     coalesce_byte_size: usize,
+// ) -> BoxFuture<'a, parquet::errors::Result<Vec<Bytes>>> {
+//     let fetch_ranges = merge_ranges(&ranges, coalesce_byte_size);
+
+//     let fetched: Vec<_> = futures::stream::iter(fetch_ranges.iter().cloned())
+//         .map(move |range| get_bytes(url, client, range))
+//         .buffered(10)
+//         .try_collect()
+//         .await?;
+
+//     todo!()
+//     // let bodies = stream::iter(fetch_ranges)
+//     //     .map(|range| {
+//     //         let client = &client;
+//     //         async move {
+//     //             let resp = client.get(url).send().await?;
+//     //             resp.bytes().await
+//     //         }
+//     //     })
+//     //     .buffer_unordered(10);
+// }
```

Comment:
Is this unused? Can we delete it? It's probably from my initial PR

Reply:
Indeed, nuke away.